File: java/MultiplesOf3Or5.java
/**
 * Tanggal: 30 April 2025
 * Link: https://www.codewars.com/kata/514b92a657cdc65150000006/train/java
 */

public class MultiplesOf3Or5 {
    public static void main(String[] args) {
        System.out.println(multiplesOf3Or5(10));
    }

    public static int multiplesOf3Or5(int number) {
        int result = 0;
        int i = 1;

        while (i < number) {
            if (i % 3 == 0 || i % 5 == 0) {
                result += i;
            }

            i++;
        }

        return result;
    }
}

File: c/multiples-of-3-or-5.c
/**
 * Tanggal: 23 Oktober 2025
 * Link: https://www.codewars.com/kata/514b92a657cdc65150000006/train/c
 */

#include <stdio.h>

int sum_of_multiples_of_3_or_5(int number) {
    if (number < 0) {
        return 0;
    }

    int sum = 0;

    for (int i = 0; i < number; i++) {
        if (i % 3 == 0 || i % 5 == 0) {
            sum += i;
        }
    }

    return sum;
}

int main() {
    printf("%d\n", sum_of_multiples_of_3_or_5(10));
    printf("%d\n", sum_of_multiples_of_3_or_5(-5));
    printf("%d\n", sum_of_multiples_of_3_or_5(98));

    return 0;
}
